metastore/src/java/org/apache/hadoop/hive/metastore/Warehouse.java

```diff
@@ -220,6 +220,18 @@ public boolean deleteDir(Path f, boolean recursive, boolean ifPurge) throws Meta
     return fsHandler.deleteDir(fs, f, recursive, ifPurge, conf);
   }
 
+  public void recycleDirToCmPath(Path f, boolean ifCopy, boolean ifPurge) throws MetaException {
+    // ifCopy input is just a place holder to support copy of data files to cmroot.
+    // Currently we through exception if it is set to true.
+    // This flag is useful in case of rename where the data won't be deleted and hence should be copied instead of move.
+    // Shall remove this exception once it is supported.
+    if (ifCopy) {
```

Comment:
We can add an optional ifCopy param to the thrift request object later when that is implemented.